api/v1/serializers/members-serializer.js

```diff
@@ -0,0 +1,86 @@
+
+const appRoot = require('app-root-path');
+const decamelize = require('decamelize');
+const JsonApiSerializer = require('jsonapi-serializer').Serializer;
+const _ = require('lodash');
+
+const { serializerOptions } = appRoot.require('utils/jsonapi');
+const { openapi } = appRoot.require('utils/load-openapi');
+const { apiBaseUrl, resourcePathLink, paramsLink } = appRoot.require('utils/uri-builder');
+
+const memberResourceProp = openapi.definitions.MemberResource.properties;
+const memberResourceType = memberResourceProp.type.enum[0];
+const memberResourceKeys = _.keys(memberResourceProp.attributes.properties);
+const memberResourcePath = 'members';
+const memberResourceUrl = resourcePathLink(apiBaseUrl, memberResourcePath);
+
+/**
+ * The column name getting from database is usually UPPER_CASE.
+ * This block of code is to make the camelCase keys defined in openapi.yaml be
+ * UPPER_CASE so that the serializer can correctly match the corresponding columns
+ * from the raw data rows.
+ */
+_.forEach(memberResourceKeys, (key, index) => {
+  memberResourceKeys[index] = decamelize(key).toUpperCase();
+});
+
+/**
+ * @summary Serialize memberResources to JSON API
+ * @function
+ * @param {[Object]} rawMembers Raw data rows from data source
+ * @param {Object} query Query parameters
+ * @returns {Object} Serialized memberResources object
+ */
+const serializeMembers = (rawMembers, query) => {
+  /**
+   * Add pagination links and meta information to options if pagination is enabled
+   */
+
+  // const pageQuery = {
+  //   size: query['page[size]'],
+  //   number: query['page[number]'],
+  // };
+
+  // const pagination = paginate(rawMembers, pageQuery);
+  // pagination.totalResults = rawMembers.length;
+  // rawMembers = pagination.paginatedRows;
+
+  const topLevelSelfLink = paramsLink(memberResourceUrl, query);
+  const serializerArgs = {
+    identifierField: 'MEMBER_ID',
+    resourceKeys: memberResourceKeys,
+    resourcePath: memberResourcePath,
+    topLevelSelfLink,
+    // query: _.omit(query, 'page[size]', 'page[number]'),
+    query,
+    enableDataLinks: true,
```

Comment:
I mentioned this before, but there is a parameter called `keyForAttribute` that can be used which will handle converting field names in the database to the field names referenced by the openapi resource. https://github.com/SeyZ/jsonapi-serializer#available-serialization-option-opts-argument. Since the names in the database don't exactly line up with the names in the specification, you will need to write a function to handle this.

This function should basically do the reverse of what the code here https://github.com/osu-mist/poker-api/blob/8b40a2fdc8e5bcffaac0f643bf90def144f109a4/api/v1/serializers/members-serializer.js#L17-L25 ends up doing.